gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAChallengeEndpoint.java

```diff
@@ -321,16 +335,19 @@ private Handler<AsyncResult<Void>> verifyHandler(RoutingContext routingContext,
 
                     cleanSession(routingContext);
                     updateStrongAuthStatus(routingContext);
+                    updateAuditLog(MFA_ENROLL_SUCCEED, endUser, client, factor, factorContext, null);
                     redirectToAuthorize(routingContext, client, endUser);
                 });
             } else {
+                updateAuditLog(MFA_CHALLENGE_SUCCEED, endUser, client, factor, factorContext, null);
                 updateStrongAuthStatus(routingContext);
                 redirectToAuthorize(routingContext, client, endUser);
             }
         };
     }
 
-    private void handleFido2Factor(RoutingContext routingContext, Client client, User endUser, String code, AsyncResult<Void> h) {
+    private void handleFido2Factor(RoutingContext routingContext, Client client, User endUser, String code,
+                                   Factor factor, FactorContext factorContext, AsyncResult<Void> h) {
         final String userId = endUser.getId();
         final JsonObject webauthnResp = new JsonObject(code);
         final String credentialId = webauthnResp.getString("id");
```

Comment:
question: Do we also have to log a Failure at line 357 🤔 ? Don't know if it make sens but looks like we will miss an audit if this error happen

Reply:
Same as before, let's talk for couple of mns during standup.

Reply:
done